### PR TITLE
Notifications timezone #649

### LINKED ISSUE #649: Notifications timezone
It is possible want to receive notifications in your local time. I believe that it should be the same as dashboard. User would set his/her preferred time.

### PR DESCRIPTION
**Pull request by VargBurz**
Closes #649 

Notification message has wrong time info due to the time difference.

## Changes:

- add optinal field in config "TIMEZONE_UTC_OFFSET", where you can specify any timezone for getting notifications.
- if there is no field, server will be used default value equals your local timezone offset.
- If the field exists, but has wrong format, there will be an error and server will not start.
- When notification message creates, `from` and `to` timestamps replace for utc time and than add utc_offset.

### CODE REVIEW

File: server/src/config.ts [hunk 1]
```diff
@@ -50,6 +51,7 @@ export const HASTIC_WEBHOOK_URL = getConfigField('HASTIC_WEBHOOK_URL', null);
 export const HASTIC_WEBHOOK_TYPE = getConfigField('HASTIC_WEBHOOK_TYPE', 'application/json');
 export const HASTIC_WEBHOOK_SECRET = getConfigField('HASTIC_WEBHOOK_SECRET', null);
 export const HASTIC_WEBHOOK_IMAGE_ENABLED = getConfigField('HASTIC_WEBHOOK_IMAGE', false);
+export const TIMEZONE_UTC_OFFSET = getTimeZoneOffset('TIMEZONE_UTC_OFFSET');
```
Comment (amper43) [inline server/src/config.ts]: let's move ''TIMEZONE_UTC_OFFSET" literal to `getTimeZoneOffset` method

File: server/src/config.ts [hunk 2]
```diff
@@ -145,3 +147,22 @@ function getDbConfig(connectionStr: string): DBConfig {
...
+    return parseTimeZone(configTimeZone);
+  } else {
+    const serverUtcOffset = moment().utcOffset();
+    return serverUtcOffset;
+  }
+}
+
+export function parseTimeZone(timeZone: string): Number {
+  const time = _.split(timeZone, ':');
+  let minutsOffset = Math.abs(Number(time[0])) * 60 + Number(time[1]);
```
Comment (amper43) [inline server/src/config.ts]: `minutesOffset`

let's add name for number `60`

File: server/src/config.ts [hunk 3]
```diff
@@ -145,3 +147,22 @@ function getDbConfig(connectionStr: string): DBConfig {
...
+  if(configTimeZone !== null) {
+    return parseTimeZone(configTimeZone);
+  } else {
+    const serverUtcOffset = moment().utcOffset();
+    return serverUtcOffset;
+  }
+}
+
+export function parseTimeZone(timeZone: string): Number {
+  const time = _.split(timeZone, ':');
```
Comment (amper43) [inline server/src/config.ts]: let's check `timeZone` have correct format

File: server/src/config.ts [hunk 4]
```diff
@@ -145,3 +148,27 @@ function getDbConfig(connectionStr: string): DBConfig {
...
+  let configTimeZone = getConfigField('TIMEZONE_UTC_OFFSET', null);
+  if(configTimeZone !== null) {
+    return parseTimeZone(configTimeZone);
+  } else {
+    const serverUtcOffset = moment().utcOffset();
+    return serverUtcOffset;
+  }
+}
+
+export function parseTimeZone(timeZone: string): number {
```
Comment (jonyrock) [inline server/src/config.ts]: Not sure it's a right place for this function. Maybe utils ?

File: server/src/services/alert_service.ts [hunk 5]
```diff
@@ -2,12 +2,19 @@ import { sendNotification, MetaInfo, AnalyticMeta, WebhookType, Notification } f
...
 import { availableReporter } from '../utils/reporter';
-import { ORG_ID, HASTIC_API_KEY, HASTIC_WEBHOOK_IMAGE_ENABLED } from '../config';
+import { ORG_ID, HASTIC_API_KEY, HASTIC_WEBHOOK_IMAGE_ENABLED, TIMEZONE_UTC_OFFSET } from '../config';
 
 import axios from 'axios';
 import * as _ from 'lodash';
+import * as moment from 'moment';
 
 
+export function toTimeZone(time: moment.MomentInput): string {
```
Comment (jonyrock) [inline server/src/services/alert_service.ts]: move it to utils

**Pull request merged**